boards/peregrine/sam4l_wm400_cape/Kconfig.defconfig

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2020-2024 Gerson Fernando Budke <nandojve@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_SAM4L_WM400_CAPE
+
+if NETWORKING
+
+config IEEE802154_RF2XX
+	default y
+	depends on IEEE802154
+
+endif # NETWORKING
+
+# This board have uses BOSSA bootloader and require that zephyr
+# relocate to code_partition.
+config USE_DT_CODE_PARTITION
+	default y
```

Comment:
You want that I create a single "Kconfig" file ? If yes, what is the difference?

Reply:
A defconfig file is for defaults, you said in the other comment that this option is required, so it's not a default because you can disable a default, therefore the board must select it

Reply:
True, what happen if someone wants to use the full space of flash with J-Link?

Reply:
If that's supported then it can remain in defconfig but from your earlier message it sounded like that is not possible?

> This board uses a SAM-BA bootloader in flash and require this option to relocate code to the correct place.

Reply:
> If that's supported then it can remain in defconfig but from your earlier message it sounded like that is not possible?
> 
> > This board uses a SAM-BA bootloader in flash and require this option to relocate code to the correct place.

The boards by default have a SAM-BA bootloader in flash. But similar to any other flash bootloader, like MCUboot, the user can use or not. The advantage of MCUboot is that flags are already handled by the infrastructure but require a flash driver in tree, which this SoC still don't have.

Are you ok if I change the message to 

`By default this board uses a SAM-BA bootloader in flash and require this option to relocate code to the correct place`